src/binary_deserializer.ml

```diff
@@ -0,0 +1,359 @@
+                      { modul
+                      ; name
+                      ; assigned_name = None
+                      ; desc = (mut, val_type)
+                      } ) )
+            | _ -> None )
+          imports
+    in
+    { Named.values; named = String_map.empty }
+  in
+  let func =
+    let values =
+      let id = ref (-1) in
+      List.map2
+        (fun typeidx (locals, body) ->
+          incr id;
+          Indexed.return !id
+            (Runtime.Local
+               { type_f = Array.get block_type_list_type typeidx
+               ; locals
+               ; body
+               ; id = None
+               } ) )
+        typeidx_list_func locals_code_list
+    in
+    let id = ref (List.length locals_code_list - 1) in
+    let values =
+      values
+      @ List.filter_map
+          (fun (modul, name, type_import) ->
+            match type_import with
+            | Func typeidx ->
+              incr id;
+              Some
+                (Indexed.return !id
+                   (Runtime.Imported
+                      { modul
+                      ; name
+                      ; assigned_name = None
+                      ; desc = Array.get block_type_list_type typeidx
+                      } ) )
+            | _ -> None )
+          imports
+    in
+    { Named.values; named = String_map.empty }
+  in
+  let table =
+    let id = ref (-1) in
+    let values =
+      List.filter_map
+        (fun (modul, name, type_import) ->
+          match type_import with
+          | Table (limits, ref_type) ->
+            incr id;
+            Some
+              (Indexed.return !id
+                 (Runtime.Imported
+                    { modul
+                    ; name
+                    ; assigned_name = None
+                    ; desc = (limits, ref_type)
+                    } ) )
+          | _ -> None )
+        imports
+    in
+    { Named.values; named = String_map.empty }
+  in
+  let data =
+    let values = List.mapi (fun id data -> Indexed.return id data) datas in
+    { Named.values; named = String_map.empty }
+  in
+  let empty_exports = { global = []; mem = []; table = []; func = [] } in
+  let exports =
+    List.fold_left
+      (fun (exports : exports) (export_typeidx, export) ->
+        match export_typeidx with
+        | '\x00' ->
+          let func = export :: exports.func in
+          { exports with func }
+        | '\x01' ->
+          let table = export :: exports.table in
+          { exports with table }
+        | '\x02' ->
+          let mem = export :: exports.mem in
+          { exports with mem }
+        | '\x03' ->
+          let global = export :: exports.global in
+          { exports with global }
+        | _ -> failwith "deserialize_exportdesc error" )
+      empty_exports exports
+  in
+  let modul =
+    { modul with global; mem; elem; func; table; start; data; exports }
+  in
+  Ok modul
+
+let empty_modul () : Simplified.modul =
```

Comment:
Could you define this as a value in `simplified.ml` directly ? `Simplified.empty_modul` for instance. We could probably use it in other parts of Owi. :)